File: BackEnd/src/domain/planLog.ts
import {
  CurriculumItemId,
  PlanId,
  PlanItemId,
  PlanLogId,
  SessionId,
} from './ids';

export enum PlanLogActionType {
  PLAN_GENERATED = 'PLAN_GENERATED',
  ITEM_COMPLETED = 'ITEM_COMPLETED',
  ITEM_POSTPONED = 'ITEM_POSTPONED',
  ITEM_CANCELLED = 'ITEM_CANCELLED',
  ITEM_TIME_UPDATED = 'ITEM_TIME_UPDATED',
  ITEM_REINSERTED = 'ITEM_REINSERTED',
  PLAN_REGENERATED = 'PLAN_REGENERATED',
}

export class PlanLog {
  planLogId: PlanLogId;
  planId: PlanId;
  sessionId?: SessionId;
  planItemId?: PlanItemId;
  curriculumItemId?: CurriculumItemId;
  actionType: PlanLogActionType;
  description: string;
  createdAt: Date;
  metadata?: Record<string, unknown>;
}

File: Frontend/GP/Models/PlanLog.cs
using System.ComponentModel.DataAnnotations;

namespace GP.Models
{
    public class PlanLog : BaseEntity
    {
        [Key]
        public int PlanLogId { get; set; }
        public int PlanId { get; set; }
        public Plan Plan { get; set; }
        public int? SessionId { get; set; }
        public Session Session { get; set; }
        public int? PlanItemId { get; set; }
        public PlanItem PlanItem { get; set; }
        public int? CurriculumItemId { get; set; }
        public CurriculumItem CurriculumItem { get; set; }
        public string ActionType { get; set; }
        public string Description { get; set; }
        public DateTime CreatedAt { get; set; }
        public string? Metadata { get; set; }
    }
}
